app/src/androidTest/java/com/sdp/movemeet/view/profile/EditProfileActivityImagePickerTest.java

```diff
@@ -21,46 +30,83 @@
 import java.io.FileNotFoundException;
 import java.io.FileOutputStream;
 import java.io.IOException;
+import java.util.concurrent.CountDownLatch;
 
+import static androidx.test.espresso.Espresso.onView;
+import static androidx.test.espresso.assertion.ViewAssertions.matches;
+import static androidx.test.espresso.matcher.ViewMatchers.withId;
 import static junit.framework.Assert.assertTrue;
 
+// This test tests both ImageHandler.uploadImage and ImageHandler.loadImage since the later one is
+// an integral part of the first one.
 public class EditProfileActivityImagePickerTest {
 
-    private StorageReference storageReference;
-    private String imagePath = "users/ydXLPxZQMXRRVHY4oGUTggDpZvp1/profile.jpg";
+    private String userImagePath = "users/B6PwbuQT3rRIyOemWtNQu7xfXmq2/profile.jpg"; // Antho1
+
+    private FirebaseAuth fAuth;
 
     @Rule
     public IntentsTestRule<EditProfileActivity> activityRule = new IntentsTestRule<>(EditProfileActivity.class);
 
     @Before
     public void setUp() {
-        // Setting up Firebase Storage
-        storageReference = FirebaseStorage.getInstance().getReference();
-
         // Saving the mocked picked image
         savePickedImage();
+
+        CountDownLatch latch = new CountDownLatch(1);
+
+        fAuth = AuthenticationInstanceProvider.getAuthenticationInstance();
+        fAuth.signInWithEmailAndPassword("antho1@gmail.com", "123456").addOnSuccessListener(new OnSuccessListener<AuthResult>() {
+            @Override
+            public void onSuccess(AuthResult authResult) {
+                latch.countDown();
+            }
+        }).addOnFailureListener(new OnFailureListener() {
+            @Override
+            public void onFailure(@NonNull Exception e) {
+                assert(false);
+            }
+        });
+
+        try {
+            latch.await();
+        } catch (InterruptedException e) {
+            assert(false);
+        }
+
+        ActivityScenario scenario = ActivityScenario.launch(EditProfileActivity.class);
```

Comment:
Shouldn't this be inside the OnSuccess ? As you can only go to this activity once you are signed in

Reply:
Well, you might be right, but I tried to modify the test in this way and, in fact, (even though it would make more sense to me too as well) it doesn't pass anymore in case I put this line inside the `OnSuccess`... I think I will leave this test as it is for the moment and will try to come back to this later on if I manage to implement it in another way 👍.